### PR TITLE
Running cargo semver-checks with no other arguments should default to checking

### PR DESCRIPTION
**Pull request by era**
Addresses #414.

It allows users to run cargo semver-checks without using the subcommand `check-release`.

I basically moved `CheckRelease` to be a field at `SemverChecks`. 

I tried to also remove it from the `SemverChecksCommands::CheckRelease` enum by using `global = true` on all fields from `CheckRelease`, but I could not add that property to `manifest` and `workspace` fields.

I reused the tests from `feature_config` to make sure the same use cases work when user specifies the subcommand or not.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -86,7 +86,13 @@ fn main() -> anyhow::Result<()> {
...
         }
         None => {
-            anyhow::bail!("subcommand required");
+            let check: cargo_semver_checks::Check = args.check_release.into();
+            let report = check.check_release()?;
+            if report.success() {
+                std::process::exit(0)
+            } else {
+                std::process::exit(1);
+            }
```
Comment (obi1kenobi) [inline src/main.rs:95]: Could we refactor the code here a bit to avoid the duplicated logic?

Maybe something like:
```rust
let check: cargo_semver_checks::Check = args.command.unwrap_or_else(|| args.check_release.into());

let report = check.check_release()?;
// ...
```

File: src/main.rs [hunk 2]
```diff
@@ -111,8 +117,8 @@ struct SemverChecks {
     #[arg(long, global = true, exclusive = true)]
     list: bool,
 
-    #[command(flatten)]
-    verbosity: clap_verbosity_flag::Verbosity<clap_verbosity_flag::InfoLevel>,
```
Comment (obi1kenobi) [inline src/main.rs:115]: I assume we don't need `verbosity` anymore because `CheckRelease` already handles it?

File: tests/feature_config.rs [hunk 3]
```diff
@@ -17,6 +17,16 @@ impl CargoSemverChecks {
             ],
         }
     }
+    fn new(current_path: &str, baseline_path: &str) -> Self {
```
Comment (obi1kenobi) [inline tests/feature_config.rs:20]:
```suggestion

    fn new(current_path: &str, baseline_path: &str) -> Self {
```

Comment (obi1kenobi) [reply tests/feature_config.rs:20]: Just to match the surrounding style.

File: tests/feature_config.rs [hunk 4]
```diff
@@ -30,6 +40,14 @@ impl CargoSemverChecks {
 
 #[test]
 fn simple_only_explicit_feature() {
+    CargoSemverChecks::new_with_check_release_subcommand(
+        "test_crates/features_simple/new/",
+        "test_crates/features_simple/old/Cargo.toml",
+    )
+    .add_arg("--only-explicit-features")
+    .run()
+    .success();
```
Comment (obi1kenobi) [inline tests/feature_config.rs:49]: This duplication is a bit unfortunate: over time as we edit these tests and add new ones, we might update one version of the check but fail to update the other. This is a maintainability risk.

Could you think of another way to set up the test where we don't need to duplicate all the test code like this? Perhaps by doing some more refactoring of the `CargoSemverChecks` type?

**Review by obi1kenobi — COMMENTED** [review]
Love the tests, great idea to use the existing test setup to ensure this functionality works too!

Just a few minor suggestions for further improvement.

File: src/main.rs (hunk 2 above)
Comment (era) [reply src/main.rs:115]: Correct, since we are flattening CheckRelease, we have the same verbosity flag available from the same places (with subcommand and not subcommand). Clap was actually the thing that highlighted it thanks to the "duplicated option" error.

**Comment by era** [comment]
Let me know if the tests are better now, I thought of adding an `assert(fn(Assert) -> ()) -> ()` function to `CargoSemverChecks`, but I was not sure if exposing the vector wouldn't be the best move. 

Also, not sure about the way I remove the subcommand from args. Let me know if you think it could be better.

File: src/main.rs [hunk 5]
```diff
@@ -75,19 +75,15 @@ fn main() -> anyhow::Result<()> {
...
-        None => {
-            anyhow::bail!("subcommand required");
-        }
+    let check: cargo_semver_checks::Check = match args.command {
+        Some(SemverChecksCommands::CheckRelease(args)) => args.into(),
+        None => args.check_release.into(),
+    };
+    let report = check.check_release()?;
+    if report.success() {
+        std::process::exit(0)
```
Comment (obi1kenobi) [inline src/main.rs:84]:
```suggestion
        std::process::exit(0);
```
I just noticed the other branch has a `;` and this one didn't.

This was already present before you made any changes so you did the right thing by leaving it as-is. I'm familiar with the context of this code and the `;` vs not doesn't make a difference, but it's nice to be consistent :)

File: tests/feature_config.rs [hunk 6]
```diff
@@ -18,13 +18,27 @@ impl CargoSemverChecks {
...
+    }
+
+    fn run_without_subcommand(&mut self) -> Assert {
+        let mut args = self.args.clone();
+        args.remove(SUBCOMMAND_ARGS_INDEX);
+        self.command().args(&args).assert()
+    }
+
+    fn run_with_subcommand(&mut self) -> Assert {
+        self.command().args(&self.args).assert()
```
Comment (obi1kenobi) [inline tests/feature_config.rs:41]: Do we need to take `&mut self` on all of these? I'm not sure we're mutating anything on any of them.

File: tests/feature_config.rs [hunk 7]
```diff
@@ -1,14 +1,14 @@
 use assert_cmd::{assert::Assert, Command};
 
+static SUBCOMMAND_ARGS_INDEX: usize = 1;
```
Comment (obi1kenobi) [inline tests/feature_config.rs:3]: This is an implementation detail on `CargoSemverChecks`, so consider moving it into `impl CargoSemverChecks` instead. The invariant it corresponds to is the way the `args` value is constructed inside `CargoSemverChecks::new()`, so the constant should be near there as well.

Also, perhaps `const` instead of `static`? The difference is subtle and it's not a big deal either way.

**Review by obi1kenobi — COMMENTED** [review]
Nicely done! A couple of tiny tweaks and this is good to go 🚀

**Comment by era** [comment]
Thank you again @obi1kenobi for all the reviews!

File: tests/feature_config.rs (hunk 7 above)
Comment (era) [reply tests/feature_config.rs]: I think I cannot use a static inside the `impl` block, but changed it to const and moved it to the `impl CargoSemverChecks` :)

**Review by obi1kenobi — APPROVED** [review]
Awesome, let's ship it 🚀
